# 
# SVN commit 1031052 by winterz:
# 
# backport SVN commit 1028614 by winterz:
# 
# don't show events that are already over. such events cannot be considered "upcoming".
# 

Index: kontact/plugins/korganizer/summaryeventinfo.cpp
===================================================================
--- kontact/plugins/korganizer/summaryeventinfo.cpp     (révision 1031051)
+++ kontact/plugins/korganizer/summaryeventinfo.cpp     (révision 1031052)
@@ -107,7 +107,8 @@
   events.setAutoDelete( true );
   KDateTime qdt;
   KDateTime::Spec spec = KPIM::KPimPrefs::timeSpec();
-  QDate currentDate = QDate::currentDate();
+  KDateTime currentDateTime = KDateTime::currentDateTime( spec );
+  QDate currentDate = currentDateTime.date();

   // prevent implicitely sharing while finding recurring events
   // replacing the QDate with the currentDate
@@ -169,6 +170,10 @@
          ( currentDate > ev->dtStart().date() || !firstDayOfMultiday ) ) {
       continue;
     }
+    // If the event is already over, then it isn't upcoming. so don't print it.
+    if ( !ev->allDay() && currentDateTime > ev->dtEnd() ) {
+      continue;
+    }

     SummaryEventInfo *summaryEvent = new SummaryEventInfo();
     eventInfoList.append( summaryEvent );
